service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadDocumentsController.java

```diff
@@ -29,8 +34,18 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
 
         CaseData caseDataBefore = getCaseDataBefore(callbackrequest);
 
+        List<Element<ApplicationDocument>> currentDocuments = Optional.ofNullable(
+            caseData.getApplicationDocuments())
+            .orElseThrow(() -> new IllegalStateException(
+                "Unexpected null current application documents. " + caseData));
+
+        List<Element<ApplicationDocument>> previousDocuments = Optional.ofNullable(
+            caseDataBefore.getApplicationDocuments())
+            .orElseThrow(() -> new IllegalStateException(
+                "Unexpected null previous application documents. " + caseDataBefore));;
+
```

Comment:
Rather than an exception which just comes up with the generic callback message, should we swap this out for returning errors to the user telling them they might need to restart the flow?